File: app/javascript/channels/favorite_channel.js
import consumer from "channels/consumer"

 // appRoomという定数に格納
consumer.subscriptions.create("FavoriteChannel", {
  connected() {
      return alert('connected');
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
      return alert(data['post']);
  },

});

File: app/channels/favorite_channel.rb
class FavoriteChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'favorite_channel'
    puts("subscribed")
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak
    ActionCable.server.broadcast('favorite_channel', {post: data['post']})
  end
end
